tiledb/sm/array/array.h

```diff
@@ -767,6 +767,11 @@ class Array {
       std::unordered_map<std::string, std::pair<uint64_t, uint64_t>>*
           max_buffer_sizes_) const;
 
+  /**
+   * Load non-remote array metadata.
+   */
+  void load_local_metadata();
```

Comment:
I don't like this name at all. The metadata isn't local; it's usually in a remote object store. Something like `do_load_metadata` or `perform_load_metadata` or `load_metadata_action` would be more descriptive.